bn_mp_shrink.c

```diff
@@ -7,20 +7,21 @@
 int mp_shrink(mp_int *a)
 {
    mp_digit *tmp;
-   int used = 1;
+   int alloc = MP_PREC;
```

Comment:
this still shrink's only down to `MP_PREC` and not the required storage for a `long long`

Reply:
I think the requirement of MP_PREC should be that it can store at least a long long.   That's what the change in tommath.h was meant for!   mp_init() also starts with allocating MP_PREC digits.

Reply:
So, for MP_8BIT, the minimum MP_PREC=16, for MP_16BIT, the minimum MP_PREC=8, for MP_32BIT the minimum MP_PREC=4, for MP_64BIT it is MP_PREC=2.   That should be documented somewhere.

Reply:
@czurnieden what do you say?

Reply:
What about adding a MP_MIN_PREC setting to the minimums given by @nijtmans and changing MP_PREC such that it is always larger than MP_MIN_PREC? 
Furthermore I think MP_PREC is usually too large. But this depends heavly on the use case. Via static_assert it could be checked that the condition always holds if MP_PREC is set by the user.

Reply:
How about an assert somewhere in the code checking that MP_PREC * sizeof(mp_digit) >= sizeof(long long)   ???

Reply:
Hm...   I meant MP_PREC * DIGIT_BIT >= sizeof(long long).    That means that MP_PREC for MP_8BIT could even be as low as 10, although that's not a power of 2.

Reply:
Hm  ....    MP_PREC * DIGIT_BIT >= CHAR_BIT * sizeof(long long)

Reply:
> How about an assert somewhere in the code checking that MP_PREC * sizeof(mp_digit) >= sizeof(long long) ???

no we need to make it sure at compile time

I'd make it `#define MP_MIN_PREC ((CHAR_BIT * sizeof(long long))/DIGIT_BIT)` and hope that nobody uses this macro in a `#if` preprocessor statement :)

and please keep the patch to to tommath.h regarding MP_8BIT

Reply:
> czurnieden what do you say?

A `MP_PREC` that is always of the same bitsize no matter what the `MP_xBIT`is would add nicely to the "rounding edges" part of the LTM update, yes.